app/main/presenters/search_presenters.py

```diff
@@ -43,20 +51,29 @@ def filters_for_question(question):
             'value': 'true',
         })
 
-    elif question['type'] in ['checkboxes', 'radios']:
-        for option in question['options']:
-            question_filters.append({
-                'label': option['label'],
-                'name': question['id'],
-                'id': '{}-{}'.format(
-                    question['id'],
-                    option['label'].lower().replace(' ', '-')),
-                'value': option['label'].lower(),
-            })
+    elif question['type'] in ['checkboxes', 'radios', 'checkbox_tree']:
+        _recursive_add_option_filters(question, question['options'], question_filters)
 
     return question_filters
 
 
+def _recursive_add_option_filters(question, options_list, filters_list):
+    for option in options_list:
+        presented_filter = {
+            'label': option['label'],
+            'name': question['id'],
+            'id': '{}-{}'.format(
+                question['id'],
+                option['label'].lower().replace(' ', '-')),
+            'value': option['label'].lower(),
+        }
+        if option.get('options'):
+            presented_filter['children'] = []
+            _recursive_add_option_filters(question, option.get('options', []), presented_filter['children'])
+
+        filters_list.append(presented_filter)
```

Comment:
I think the most annoying thing is that I've ended up with basically the same recursion in about a zillion different places. Replacing them all with a single "flatten tree" generator thing would be a nice thing for the future, but not on the timescales I have right now :(

(Incidentally doing it with generators is a bit nicer once we can use `yield from` which isn't in Python 2.)